### DISCUSSION TITLE
How does `matchCurrentAge` work?

### DISCUSSION THREAD

**Discussion by p3k** [renovatebot/renovate#27113] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.168.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I am trying to add labels to merge requests for packages that are older than `> 6 months`:

```json
{
  "matchCurrentAge": "> 6 months",
  "addLabels": ["Overdue"]
}
```

– but it results in a very incomprehensible, almost random assignment of the label, e.g.:

| Package | Update | Change |
|---|---|---|
| [gradle](https://gradle.org) ([source](https://github.com/gradle/gradle)) | patch | `7.6.3` -> `7.6.4` |

* Renovate authored the initial merge requeset 4 months ago
* `7.6.4` was released on 5 Feb 2024
* `7.6.4` is definitely younger than 6 months
* `7.6.3` was released 4 Oct 2023
* there's less than 6 months between the releases.

Other updates – that in fact already became stale because of some issues that need to be resolved first – did *not* get the label.

How does Renovate calculate the current age? What does it match against?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Labels are added only at PR creation time and not kept updated after that. However there's a feature request in progress to make this possible soon

**Reply by p3k** [reply]
OK but that does not seem to be the (only) issue here… 

Some merge requests clearly got the label although there is no indication of *something* is older than six months (see list items for the gradle package).

Some others which *should* get it, did *not* get it, and while this could indicate the MR was already existing, so the label was not added, it still is confusing.

So what is the deal of `matchCurrentAge` – should it even work the way I tried to apply it?

**Reply by rarkins** [reply]
It only works if the datasource itself returns release timestamps. If the datasource doesn't return timestamps we can use, then we can't calculate the current release's age. Otherwise it should work as you expect, e.g. adding a label if the current version is particularly old

**Reply by frickua** [reply]
Sorry, can not get it. I think `matchCurrentAge` documentation is unclear. Could You please explain: Can I update some gradle dependencies based on their age, I mean -  I have `aws` dependency that update to frequent, so I want to update it only if my current version outdated for  ~ 3 month. Does it possible with `matchCurrentAge` or not? If not it will be a great feature!

**Reply by rarkins** [reply]
There isn't a simple yes/no answer. It depends on whether the registry/datasource supports a way to return releaseTimestamp. For non-default maven registries (which Gradle uses), it's often not possible.

**Reply by frickua** [reply]
Thanks for the rapid answer! How can I check the support? 
I have last modified column on my proxy repo:
![image](https://github.com/user-attachments/assets/a85fa5c2-9778-411a-9920-cd98c0624f30)
and similar on maven central - 
![image](https://github.com/user-attachments/assets/b37dc1e6-d962-4148-8cee-4df1f9032efc)
Does this enough?
Also I have `DEBUG: No currentVersionTimestamp for com.amazonaws:aws-java-sdk-sts ` log. Does this 100% guaranteed that `matchCurrentAge` not supported In my case?

**Reply by rarkins** [reply]
It definitely means not supported. Feel free to raise a PR or suggest a feature request if you have a concrete idea of how to extract timestamps for your particular registry/proxy
